inst/include/dust/dust.hpp

```diff
@@ -459,11 +589,18 @@ class Dust {
   }
 
   std::vector<uint64_t> rng_state() {
+    refresh_host();
     return _rng.export_state();
   }
 
   void set_rng_state(const std::vector<uint64_t>& rng_state) {
+    // TODO: In dustgpu we don't do a refresh_host() here but I think
+    // we need to. Otherwise if the canonical state is on the device
+    // it's out of sync. However, this is an issue for other things
+    // (e.g., set_state()) so we might need to think about this more
+    // generally.
```

Comment:
Yeah I think that was an error – happy with this

Reply:
Fixed in 0e367ea